Added missing deploy settings in cloud section (#3934)
umbraco/UmbracoDocs: Umbraco-Cloud/Deployment/Deploy-Settings/index.md

# Configuration for Umbraco Deploy

> All configuration for Umbraco Forms is held in the `appSettings.json` file found at the root of your Umbraco website.  If the configuration has been customized to use another source, then the same keys and values discussed in this article can be applied there.

Reviewer: ```suggestion
All configuration for Umbraco Deploy is held in the `appSettings.json` file found at the root of your Umbraco website.  If the configuration has been customized to use another source, then the same keys and values discussed in this article can be applied there.
```
Section heading says Deploy

Author: thanks did not see that before 😅

In a similar way, Deploy can be configured to allow for backoffice transfers of dictionary items instead of using files serialized to disk, by setting `TransferDictionaryAsContent` as `true`.

## AllowMembersDeploymentOperations and TransferMemberGroupsAsContent

> As of version 9.3.0, it's also possible to transfer members and member groups via the back-office between environments.  This is disabled by default as a deliberate decision to make use of the feature needs to be taken, as for most installations it will make sense to have member data created and managed only in production. There are obvious potential privacy concerns to consider too.  However, if being able to deploy and restore this information between environments makes sense for the specific workflow of your project, it's a supported scenario.

Reviewer: Try simplifying to remove all the jargon/marketing terms.

```suggestion
This is disabled by default, as for most installations, member data is created and managed _only_ in production. If you want to deploy and restore member information between environments for specific workflows of your project, you can do so in the `appsettings.json` file
```

To enable, you can add or amend the `AllowMembersDeploymentOperations` and `TransferMemberGroupsAsContent` settings.

> 

Reviewer: How about adding the code snippet here? https://umbraco.com/blog/release-candidates-umbraco-forms-81193-and-deploy-4593/

For example, if you have a media picker on a content item, that references a media item that's been deleted or is in the recycle bin, you'll get an error and the deploy won't complete until the issue is resolved (by removing the reference to the deleted media item).

You can configure deploy to ignore these issues and proceed with the transfer operation without warning, by setting the value of this option to `true`.

## IgnoreBrokenDependenciesBehavior

For finer control of the above setting you can amend this value from the default of `All` to either `Transfer` or `Restore`.

For example, using the following settings, you will have an installation that ignores broken dependencies when restoring from an upstream environment.  It will however still prevent deployment and report any dependency issues when attempting a transfer to an upstream environment.

```json
    "IgnoreBrokenDependencies": true,
    "IgnoreBrokenDependenciesBehavior": "Restore",
```

## Memory cache reload

> Some customers have reported intermittent issues related to Umbraco's memory cache following deployments, which are resolved by a manual reload of the cache via the _Settings > Published Status > Caches_ dashboard.  If you are running into such issues and are able to accomodate a cache clear after deployment, this workaround can be automated via the following setting:

Reviewer: ```suggestion
If you are running into such issues and can accommodate a cache clear after deployment, this workaround can be automated via the following setting:
```

```json
    "ReloadMemoryCacheFollowingDiskReadOperation": true,
```

> By upgrading to the most recent available version of the CMS major you are running, you'll be able to benefit from the latest bug fixes and optimizations in this area.  That should be your first option if encountering cache related issues. Failing that, or if a CMS upgrade is not an option, then this workaround can be considered.

Reviewer: ```suggestion
By upgrading to the most recent available version of the CMS major you are running, you'll be able to benefit from the latest bug fixes and optimizations in this area.  That should be your first option if encountering cache-related issues. Failing that, or if a CMS upgrade is not an option, then this workaround can be considered.
```

## Deployment of culture & hostnames settings

Culture and hostname settings, defined per content item for culture invariant content, are not deployed between environments by default but can be opted into via configuration.

```json
    "AllowDomainsDeploymentOperations": "None|Culture|AbsolutePath|Hostname|All",
```

To enable this, set the configuration value as appropriate for the types of domains you want to allow:

- *Culture* - the language setting for the content, defined under "Culture"
- *AbsolutePath* - values defined under "Domains" with an absolute path, e.g. "/en"

> - *Hostname* - values defined under "Domains" with a full host name, e.g. "en.mysite.com"

Reviewer: ```suggestion
- *Hostname* - values defined under "Domains" with a full hostname, e.g. "en.mysite.com"
```

> Combinations of settings can be applied, e.g. `Hostname,AbsolutePath`.

Reviewer: ```suggestion
Combinations of settings can be applied, e.g. `Hostname, AbsolutePath`.
```
